sdk/core/azure_core_amqp/src/fe2o3/value.rs

```diff
@@ -162,6 +162,7 @@ impl From<AmqpValue> for fe2o3_amqp_types::primitives::Value {
             AmqpValue::Array(a) => fe2o3_amqp_types::primitives::Value::Array(
                 a.into_iter().map(|v| v.into()).collect(),
             ),
+            AmqpValue::Composite(_) => panic!("Composite values are not supported in Fe2o3"),
```

Comment:
Is panicking truly the right thing here? It'll bring down the app. Is it low likelihood at least? Can we document this? You can put docs on trait impls to override the trait docs if that would help.

Reply:
I don't see a way of returning an error in a From trait implementation.

The Fe2o3 AMQP implementation doesn't directly support the AMQP "Composite" type - instead it treats AMQP "Composite" types as a serialization artifact - basically each serializable construct has a set of composition attributes which trigger the creation of an AMQP Composite when serializing and deserializing the type. TBH, this is actually rather clever. But uAMQP *does* have Composite as a first class primitive, so for interoperability, I need to have one in the AMQP wrapper.

That works great until you try to convert an AMQP type to an Fe2o3 type. If you do that, I * want* the app to be terminated, because it's never going to work.

Reply:
There is `TryFrom`, but not sure if that makes things harder for C++ and/or the Rust UX. Even in that case, `From` could only panic.